pwndbg/aglib/regs.py

```diff
@@ -0,0 +1,296 @@
+"""
+Reading register value from the inferior, and provides a
+standardized interface to registers like "sp" and "pc".
+"""
+
+from __future__ import annotations
+
+import ctypes
+import re
+import sys
+from types import ModuleType
+from typing import Any
+from typing import Callable
+from typing import Dict
+from typing import Generator
+from typing import List
+from typing import Tuple
+from typing import cast
+
+import pwndbg
+import pwndbg.aglib.arch
+import pwndbg.aglib.proc
+import pwndbg.aglib.qemu
+import pwndbg.aglib.remote
+import pwndbg.aglib.typeinfo
+import pwndbg.lib.cache
+from pwndbg.dbg import EventType
+from pwndbg.lib.regs import BitFlags
+from pwndbg.lib.regs import RegisterSet
+from pwndbg.lib.regs import reg_sets
+
+
+@pwndbg.lib.cache.cache_until("stop", "prompt")
+def regs_in_frame(frame: pwndbg.dbg_mod.Frame) -> pwndbg.dbg_mod.Registers:
+    return frame.regs()
```

Comment:
Not sure this needs to be cached but maybe

Reply:
Caching that lets us avoid calling `SBFrame::GetRegisters` (which pulls in all register values at once) every time we want the value of a single register.